apps/examples/memory_fragmentation_test/memory_fragmentation_test.c

```diff
@@ -0,0 +1,318 @@
+	if (r < 0) {
+		print_usage();
+	}
+
+	srand(SEED);
+
+	/* Allocate memory according to 'numof_size' */
+	memory_allocation(list, numof_size, num_alloc);
+
+	/* Free the number of memory allocation according to 'num_free' 
+	 * with a first-come first-free manner.
+	 */
+	printf("Test Configuration\n");
+	printf("Size [bytes]	# of init alloc		# of free\n");
+	for (i = 0; i < NUMOF_SMALL_MEMORY; ++i) {
+		num_free[i] = num_alloc[i] * f1 / 10;
+		if (num_alloc[i] > 0) {
+			printf("%d		%d			%d\n", 1 << (i + 4), num_alloc[i], num_free[i]);
+		}
+	}
+	for (i = NUMOF_SMALL_MEMORY; i < MAX_SIZE_EXPONENT; ++i) {
+		num_free[i] = num_alloc[i] * f2 / 10;
+		if (num_alloc[i] > 0) {
+			printf("%d		%d			%d\n", 1 << (i + 4), num_alloc[i], num_free[i]);
+		}
+	}
+	memory_free(num_free, num_alloc, list);
+
+	/* Repeat a cycle of allocation and free a given number of times */
+	for (i = 0; i < r; ++i) {
+		/* memory allocation according to 'num_free' */
+		for (j = 0; j < MAX_SIZE_EXPONENT; ++j) {
+			num_alloc_tmp[j] = 0;
+		}
+		memory_allocation(list, num_free, num_alloc_tmp);
+		for (j = 0; j < MAX_SIZE_EXPONENT; ++j) {
+			num_alloc[j] += num_alloc_tmp[j];
+		}
+		
+		/* free allocation according to 'num_free' */
+		memory_free(num_alloc_tmp, num_alloc, list);
+	}
+
+	printf("\nMemory allocation & free repeated at %d times.\n", r);
+
+	/* wait for 10 seconds before free-ing all the memory */
+	printf("\nAll memory test has been done.\n");
+	printf("Memory allocation status: \n");
+	printf("Size [bytes]	# of alloc/free(repeat)	# of remaining alloc\n");
+	for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+		if (num_alloc[i] > 0) {
+			printf("%d		%d			%d\n", 1 << (i + 4), num_free[i], num_alloc[i]);
+		}
+	}
+	printf("\nWe are going to wait for 10 seconds before free-ing all the memory.\n");
+	printf("Please, type 'heapinfo' to check how the heap memory is fragmented.\n");
+	sleep(10);
+
+	/* Free all the memory */
+	for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+		next[i] = list[i].next;
+	}
+	for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+		while (next[i]) {
+			item = next[i];
+			free(item->data);
+			next[i] = next[i]->next;
+			free(item);
+			--num_alloc[i];
+		}
+	}
+
+	i = 0;
+	for (j = 0; j < MAX_SIZE_EXPONENT; ++j) {
+		i += num_alloc[j];
+	}
+
+	printf("\nTen seconds have passed.\n");
+	if (i == 0) {
+		printf("All the memory is released.\n");
+	} else {
+		printf("All the memeory can't be released. # of allocation: %d\n", i);
+		for (j = 0; j < MAX_SIZE_EXPONENT; ++j) {
+			if (num_alloc[j] > 0) {
+				printf("%d [Bytes] : %d\n", 1 << (j + 4), num_alloc[j]);
+			}
+		}
+	}
+}
+
+static int test_task(int argc, char *argv[])
+{
+	memory_fragmentation_test(argc, argv);
+
+	task_delete(0);
+	return 0;
+}
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int memfrag_main(int argc, char *argv[])
+#endif
+{
+	printf("Memory fragmentation test!!\n");
+	task_create("memory fragmentation test", 100, 2048, test_task, argv);
+
```

Comment:
Corresponding to failure or success of test case, you can add a success/failure log fo rthis TC

Reply:
This is not a testcase to see success or failure, but a test application to intentionally fragment the heap memory.